### DISCUSSION TITLE
Reading digital meter texts

### DISCUSSION THREAD

**Discussion by Trainee2024** [mindee/doctr#1520] [Q&A]
Hello , I am new to OCR and using docTR for reading text from digital meter image .  I am not getting proper output , I also get warning like this "Fontconfig error: Cannot load default config file"  Any help would be appreciated . Thank you

I am attaching the image i am using . I am using this code : 

from doctr.models import ocr_predictor
from doctr.io import DocumentFile

model = ocr_predictor(det_arch='db_mobilenet_v3_large', reco_arch='crnn_vgg16_bn', pretrained=True)

single_img_doc = DocumentFile.from_images("Cropped.png")

result = model(single_img_doc)
text_output = result.render()
print(text_output)
========= Output =======
GJ1474
No.
Serial
1
10
23:55:54
2024
Jan
ter
Unit Regist
VArh
1411107

**Comment by Trainee2024** [comment]
My image is 
![Uploading Cropped.png…]()

**Reply by felixdittrich92** [reply]
Hi @Trainee2024 👋,
Looks like the Image isn't uploaded correct could you try it again please :)

If your image contains more line like elements you can disable resolving blocks by passing `resolve_blocks=False` to the `ocr_predictor` instance.

**Comment by Trainee2024** [comment]
![demo](https://github.com/mindee/doctr/assets/164501840/8956223c-09b0-4151-a199-ed8ea6721536)

**Comment by Trainee2024** [comment]
Here, I m attaching other image for your reference, I want to create script which reads from the green box.
![vlcsnap-2024-02-29-17h59m21s426](https://github.com/mindee/doctr/assets/164501840/8daaee1e-e47c-48ab-b9d7-7622345dc3b2)

**Reply by felixdittrich92** [reply]
Hi @Trainee2024 :wave:,
You can process the whole image and extract afterwards the field from coords with the required values or try to pass the crop directly.
I suggest to pass `assume_straight_pages=False` to the `ocr_predictor` instance so the minor rotations will be also handled.
As recognition model i used `parseq` (you need to test a bit the different available recognition models which fits best your needs)
BTW you can further improve the detection results by playing a bit with some advanced features: 
https://mindee.github.io/doctr/using_doctr/using_models.html#advanced-options

Ref:
![Screenshot from 2024-03-26 07-41-32](https://github.com/mindee/doctr/assets/23098070/5508bb24-053a-4408-a35b-121e5c056118)

**Comment by Trainee2024** [comment]
Thank you for the prompt response . I really appreciate it. I did try the detection and getting the Image as you have attached . Here is my output image : 
![Sample Detection](https://github.com/mindee/doctr/assets/164501840/f5f5b71f-daf5-421a-8b4b-f073f2926a3e)

But while extracting the texts , it is not giving the accurate result . Some values are missing .Do you have suggestion for this type of meter reading fonts images ? 
Thank you in advance

**Reply by felixdittrich92** [reply]
I think in this case the only option would be to fine tune your own custom recognition model because our models was trained (mostly) on document image word crops and your image goes more into the direction of scene text recognition in the wild so i think all our models will have some problems with it.
If you use the crop from the required area the detection / recognition is still worse ?
training references/docs: https://mindee.github.io/doctr/using_doctr/custom_models_training.html

**Reply by Trainee2024** [reply]
Yes , Even after cropping image , we are still getting inconsistent result . 
![Cropped](https://github.com/mindee/doctr/assets/164501840/62114fd0-cfc9-43f7-8e6e-010f9084762d)
 Is there any way  OR any document / youtube video , how can we  fine tune our own custom recognition model  ? I am new to it and  this  https://mindee.github.io/doctr/using_doctr/custom_models_training.html looks confusing for me. 

Thank you

**Reply by felixdittrich92** [reply]
Unfortunately no, that's maybe something for mindee internal content creator.
I will forward this request :+1:

**Reply by Trainee2024** [reply]
Thank you so much @felixdittrich92 . 
Looking forward to any assistance . If anyone on the team has insight into how to extract text from the meter image, we would greatly appreciate your help. Thank you.

**Reply by felixdittrich92** [reply]
Forwarded :)
You can also join the mindee doctr slack channel to ask for more help or about the tutorial progress:
https://join.slack.com/t/mindee-community/shared_invite/zt-2fkp541cy-4QNb8Mlbhs8Gh8hPBe8kqw
